### PR TITLE
Add Processes to metadata.

### PR DESCRIPTION
**Pull request by wwitzel3**
This PR adds Processes information to the charm metadata. This information will be used by the workload process management feature of Juju.

### CODE REVIEW

File: process.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+func (p Process) Validate() error {
+	if p.Name == "" {
+		return fmt.Errorf("missing name")
+	}
+	if p.Type == "" {
+		return fmt.Errorf("metadata: processes.%s.type: name is required", p.Name)
+	}
+
+	for _, volume := range p.Volumes {
+		//panic(fmt.Sprintf("%#v", volume))
```
Comment (cherylj) [inline process.go]: Delete?

File: meta.go [hunk 2]
```diff
@@ -426,6 +428,10 @@ func (meta Meta) Check() error {
 		names[name] = true
 	}
 
+	if err := checkProcesses(meta.Processes); err != nil {
```
Comment (davecheney) [inline meta.go]: return checkProcess(meta.Processes)

File: process.go [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+// ProcessVolume is storage volume information for a workload process.
+type ProcessVolume struct {
+	// ExternalMount is the path on the host.
+	ExternalMount string
+	// InternalMount is the path on the process.
+	InternalMount string
+	// Mode is the "ro" OR "rw"
+	Mode string
+	// Storage is the name the metadata entry, if any.
+	Storage string
```
Comment (davecheney) [inline process.go]: s/Storage/Name

**Comment by wwitzel3** [comment]
$$merge$$

**Pull request merged**